### ISSUE TITLE
Failing tests with intel mpi

### ISSUE THREAD

**Issue by fsimonis** [precice/precice#966]
**Describe your setup**

preCICE Version: 2.2.0
Python 3.8.2
impi: 2019.7.217
iccifort: 2020.1.217
intel: 2020a

**Describe the problem**

The following tests fail:
```
1 - precice.acceleration (Failed)
4 - precice.com.mpiports (Failed)
8 - precice.m2n.mpiports (Failed)
15 - precice.serial (Failed)
16 - precice.parallel (Failed)
```

They fail with something along these lines:
```
MPI startup(): unsupported scenario. mlx provider does not support Spawns by default, please set FI_MLX_ENABLE_SPAWN=yes.
15: 
15: Abort(805944080) on node 1 (rank 1 in comm 0): Fatal error in PMPI_Open_port: Internal MPI error!, error stack:
15: PMPI_Open_port(120): MPI_Open_port(MPI_INFO_NULL, port=0x14b1aa0) failed
15: MPID_Open_port(198): 
15: (unknown)(): Internal MPI error!
```

**Expected behaviour**
If there was a problem with the ports-implementation, then the only the mpiports test suites should fail.
At least the serial, parallel and acceleration tests should run.

**Additional context**
Related to https://github.com/easybuilders/easybuild-easyconfigs/pull/12270

**Comment by besserox** [comment]
Attaching here the full logfiles of the build and test execution failures on 2 different clusters, but same version of preCICE and MPI.

### On Barbora cluster (IT4I):

```
62% tests passed, 11 tests failed out of 29

Label Time Summary:
Solverdummy    =   9.67 sec*proc (9 tests)
mpiports       =   0.55 sec*proc (2 tests)
petsc          =   0.43 sec*proc (1 test)

Total Test time (real) =  21.90 sec

The following tests FAILED:
          1 - precice.acceleration (Failed)
          4 - precice.com.mpiports (Failed)
          8 - precice.m2n.mpiports (Failed)
         15 - precice.serial (Failed)
         16 - precice.parallel (Failed)
         24 - precice.solverdummy.run.cpp-cpp (Failed)
         25 - precice.solverdummy.run.c-c (Failed)
         26 - precice.solverdummy.run.fortran-fortran (Failed)
         27 - precice.solverdummy.run.cpp-c (Failed)
         28 - precice.solverdummy.run.cpp-fortran (Failed)
         29 - precice.solverdummy.run.c-fortran (Failed)
```

[easybuild-preCICE-2.2.0-20210225.110134.xdviR.log](https://github.com/precice/precice/files/6042153/easybuild-preCICE-2.2.0-20210225.110134.xdviR.log)


### On Iris cluster (Uni.lu)

```
83% tests passed, 5 tests failed out of 29

Label Time Summary:
Solverdummy    =  29.74 sec*proc (9 tests)
mpiports       =   1.13 sec*proc (2 tests)
petsc          =   0.73 sec*proc (1 test)

Total Test time (real) =  52.74 sec

The following tests FAILED:
          1 - precice.acceleration (Failed)
          4 - precice.com.mpiports (Failed)
          8 - precice.m2n.mpiports (Failed)
         15 - precice.serial (Failed)
         16 - precice.parallel (Failed)
```

[easybuild-preCICE-2.2.0-20210225.110420.nMBJq.log](https://github.com/precice/precice/files/6042154/easybuild-preCICE-2.2.0-20210225.110420.nMBJq.log)

**Comment by fsimonis** [comment]
Resolved by #1826
